core/src/main/scala/org/apache/spark/util/random/RandomSampler.scala

```diff
@@ -107,6 +113,28 @@ class BernoulliCellSampler[T](lb: Double, ub: Double, complement: Boolean = fals
 
   override def setSeed(seed: Long): Unit = rng.setSeed(seed)
 
+  override def sample(): Int = {
```

Comment:
Can we not have the code duplication and write the iterator version based on this function?